index.html

```diff
@@ -257,21 +296,125 @@ <h1>Settings</h1>
                 if (!response) {
-                    loadJSON("https://cdn.emulatorjs.org/versions.json", resp => loadVersions(resp));
+                    loadJSON("https://cdn.emulatorjs.org/versions.json", (resp) =>
+                        loadVersions(resp)
+                    );
                     document.getElementById("offline-status").textContent = "Offline Status: NOT INSTALLED";
                 } else {
                     loadVersions(response);
                     document.getElementById("offline-status").textContent = "Offline Status: READY";
                 }
             });
+
+            const corePath = "https://cdn.emulatorjs.org/stable/data/cores/";
+
+            function loadCoreList() {
+                loadJSON(corePath + "cores.json", (response) => {
+                    const cacheCores = JSON.parse(localStorage.getItem("cachedCores")) || [];
+                    const cores = JSON.parse(response);
+                    document.getElementById("cores-form").innerHTML = "";
+                    for (const core of cores) {
+                        const label = document.createElement("label");
+                        const input = document.createElement("input");
+                        input.type = "checkbox";
+                        input.name = "core";
+                        input.value = core.name;
+                        if (cacheCores.some(arr => arr[0] === core.name)) {
+                            input.checked = true;
+                        }
+                        label.appendChild(input);
+                        const cachedCore = cacheCores.find(arr => arr[0] === core.name);
+                        const additionalText = cachedCore ? ` (${cachedCore[1]})` : "";
+                        label.appendChild(document.createTextNode(core.name + additionalText));
+                        document.getElementById("cores-form").appendChild(label);
+                    }
+                });  
+            }
+
+            function cacheCores() {
+                const cacheCores = JSON.parse(localStorage.getItem("cachedCores")) || [];
+                const cores = document.getElementsByName("core");
+                const stable = localStorage.getItem("latest-stable");
+                const urlsToCache = [];
+                const urlsToRemove = [];
+                for (const core of cores) {
+                    const cachedCore = cacheCores.find(arr => arr[0] === core.value);
+                    if (core.checked && (!cachedCore || cachedCore[1] !== stable)) {
+                        urlsToCache.push(corePath + "reports/" + core.value + ".json");
+                        urlsToCache.push(corePath + core.value + "-wasm.data");
+                        urlsToCache.push(corePath + core.value + "-thread-wasm.data");
+                        urlsToCache.push(corePath + core.value + "-legacy-wasm.data");
+                        urlsToCache.push(corePath + core.value + "-thread-legacy-wasm.data");
```

Comment:
There should be some sort of opt-in here. This is 4x the amount of needed storage to keep a core locally. A user isn't often going to be switching between these at all